packages/@lwc/engine/dom/src/apis/lightning-element.ts

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+import { defineProperty, freeze, isUndefined, seal } from '@lwc/shared';
+
+import {
+    BaseLightningElement,
+    buildCustomElementConstructor,
```

Comment:
The entire `buildCustomElementConstructor` method should be extracted in the `engine-dom` since it something that is only usable in the DOM context.

Reply:
Yeah, but I think it's better to do that in the renderer PR because this requires a lot of changes that have already been done in that branch.

Reply:
Ok sure let that that in.